src/os/FileJournal.cc

```diff
@@ -61,6 +61,10 @@ int FileJournal::_open(bool forwrite, bool create)
       int err = errno;
       derr << "FileJournal::_open: error closing old fd: "
 	   << cpp_strerror(err) << dendl;
+      // We'd better quit in this situation, otherwise
+      // this fd is going to be replaced and therefore
+      // has no more choice to be correctly closed. 
+      return -err;
```

Comment:
Looking at the close(2) man page there are only 3 possible errors:

1. EBADF (not a valid file descriptor)  This could be a corrupt fd value.  No point in not going forward.
2. EINTR (interrupted by signal)  The TEMP_FAILURE_RETRY() macro retries as long as error is EINTR.
3. EIO (I/O error) This happens because an async write() didn't report the error, but the fd is still closed afterwards.

This change should be removed.